tests/xammac_tests/AppKit/NSGridViewTest.cs

```diff
@@ -0,0 +1,48 @@
+//
+// Unit tests for NSGridView
+//
+
+
+#if XAMCORE_2_0 && __IOS__
+
+using AppKit;
+using Foundation;
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.AppKit {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class NSGridViewTest {
+
+		[Test]
+		public void CreateWithNSViewArrayOfNSViewArray ()
+		{
+			NSView [][] nSViewsArrayOfArray = new NSView [4][];
+			nSViewsArrayOfArray [0] = new NSView [1] { new NSView() };
+			nSViewsArrayOfArray [1] = new NSView [1] { new NSView() };
+			nSViewsArrayOfArray [2] = new NSView [1] { new NSView() };
+			nSViewsArrayOfArray [3] = new NSView [1] { new NSView() };
+            
+			NSGridView nSGridViewArrayOfArray = NSGridView.Create (nSViewsArrayOfArray);
+			
+			Assert.NotNull(nSGridViewArrayOfArray);

```

Comment:
This is a great start!

However, I'd add more asserts to make sure things really work as expected:

 1. Make each NSView different, doesn't matter what exactly, it's just so that you can assert you get the right one later. For instance, make them all `NSTextView`s and set the Value property:

    ```csharp
    nSViewsArrayOfArray [0] = new NSView [1] { new NSTextView ()  { Value = "0" } };
    nSViewsArrayOfArray [1] = new NSView [1] { new NSTextView ()  { Value = "1" } };
    // ...
    ```

    And then assert that each cell is the expected one:

    ```csharp
    Assert.AreEqual ("0", ((NSTextView) nSGridViewArrayOfArray.GetCell (0, 0)).Value, "0,0");
    Assert.AreEqual ("1", ((NSTextView) nSGridViewArrayOfArray.GetCell (0, 1)).Value, "0,1");
    // ...
    ```

3. also what happens with other types of input, such as arrays of different length:

    ```csharp
    nSViewsArrayOfArray [0] = new NSView [1] { new NSTextView ()  { Value = "0" } };
    nSViewsArrayOfArray [1] = new NSView [2] { new NSTextView ()  { Value = "1" }, new NSTextView () { Value = "1bis" } };
    // ...
    ```

4. or arrays with nulls:

    ```csharp
    nSViewsArrayOfArray [0] = new NSView [1] { new NSTextView ()  { Value = "0" } };
    nSViewsArrayOfArray [1] = new NSView [1] { null } };
    // ...
    ```

5. or null arrays:

    ```csharp
    nSViewsArrayOfArray [0] = new NSView [1] { new NSTextView ()  { Value = "0" } };
    nSViewsArrayOfArray [1] = null;
    // ...
    ```

6. or even just null input altogether:


    ```csharp
    nSViewsArrayOfArray = null;
    ```

Basically try to break your own code :smile: